hyperspy/_signals/eds_tem.py

```diff
@@ -622,31 +623,9 @@ def decomposition(self,
         closing: bool
             If true, applied a morphologic closing to the maks obtained by
             vacuum_mask.
-        algorithm : 'svd' | 'fast_svd' | 'mlpca' | 'fast_mlpca' | 'nmf' |
-            'sparse_pca' | 'mini_batch_sparse_pca'
-        output_dimension : None or int
-            number of components to keep/calculate
-        centre : None | 'variables' | 'trials'
-            If None no centring is applied. If 'variable' the centring will be
-            performed in the variable axis. If 'trials', the centring will be
-            performed in the 'trials' axis. It only has effect when using the
-            svd or fast_svd algorithms
-        auto_transpose : bool
-            If True, automatically transposes the data to boost performance.
-            Only has effect when using the svd of fast_svd algorithms.
-        signal_mask : boolean numpy array
-            The signal locations marked as True are not used in the
-            decomposition.
-        var_array : numpy array
-            Array of variance for the maximum likelihood PCA algorithm
-        var_func : function or numpy array
-            If function, it will apply it to the dataset to obtain the
-            var_array. Alternatively, it can a an array with the coefficients
-            of a polynomial.
-        polyfit :
-        reproject : None | signal | navigation | both
-            If not None, the results of the decomposition will be projected in
-            the selected masked area.
+        **kwargs
```

Comment:
It would be convenient for the user to have all the parameters described here without having to look at the docstring of `~.learn.mva.MVA.decomposition`. Shall we define the parameters docstring in `~.docstring.mva` and reuse it in `~.learn.mva.MVA.decomposition` and `here?

Reply:
Let me some minutes to have a quick look, it's a good distraction from my current boring tasks...

Reply:
Agreed I wasn't sure of the best way forward

Reply:
This is the only place it's wrapped for example, because it sets different defaults for Poisson noise compared to the standard func

Reply:
Last commit copies over the parameters to the docstring.